service.go

```diff
@@ -107,3 +107,22 @@ func NewServiceEntry(instance, service, domain string) *ServiceEntry {
 		ServiceRecord: *NewServiceRecord(instance, service, domain),
 	}
 }
+
+// ServiceEventType includes the ServiceEntry along with the event that happened in Network
+type ServiceEventType int
+
+const (
+	NewOrUpdated ServiceEventType = iota
```

Comment:
Please make `New` and `Updated` two separate event types.
Further, we could introduce a bit-wise mapping, so it is easy to filter the requests. e.g.
```
const (
	New ServiceEventType = 1 << (iota),
	// etc
)
```